pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -90,6 +94,11 @@ func (h *handler) getArgsEnvAndStatus(infraObj *infraObject, args map[string]int
 		},
 	}
 
+	machine, err := h.machines.Get(infraObj.meta.GetNamespace(), infraObj.meta.GetName())
+	if err != nil && !apierror.IsNotFound(err) {
```

Comment:
I don't think we want to continue if the machine isn't found.

Reply:
agreed, fixed this